airflow/providers/odbc/hooks/odbc.py

```diff
@@ -212,17 +216,27 @@ def get_sqlalchemy_connection(
         cnx = engine.connect(**(connect_kwargs or {}))
         return cnx
 
-    @staticmethod
-    def _make_serializable(result: list[pyodbc.Row] | pyodbc.Row | None) -> list[NamedTuple] | None:
-        """Transform the pyodbc.Row objects returned from an SQL command into JSON-serializable NamedTuple."""
+    def _make_serializable(self, result: Any) -> Any:
```

Comment:
Actually what we would have to do, to not yank the current odbc provider is to add code in the common.sql to check if it's there and call it. Adding it here while we also add >= 1.9.1 on the common.sql makes no sense (because 1.9.1 will never call it)

The back-compatibilty here is to let the "just released" odbc provider to work with the new `"common.sql" - not the other way round.

But maybe better - just remove it and I will simply yank the odbc provider we just released, it's going to be way simpler @Joffreybvn

Reply:
I think we need this back-compat fix anyway because the `make_serializable´ was introduced in odbc two patches ago (in [4.2.0](https://airflow.apache.org/docs/apache-airflow-providers-odbc/4.3.0/changelog.html#id2)).

Thus, I added it with a warning. Airflow 2.8.1 constraint will need this latest odbc provider version to not annoy users with the warning. So that only people who stick with odbc==4.2.0 or odbc==4.3.0 while upgrading common.sql get warned.